Yuzu no longer compiles in linux (gcc 7.3.0, compilation error in boost/icl)

On test system (Linux Mint 19.2 x64, 4.18 kernel, gcc 7.3.0), and as of 41566c615b86183c4dbec5a6bc7596ed4d8a116b, neither yuzu nor yuzu-canary compiles anymore.  Last successful compile was on Feb 25th (master-95c37df87) but have not tried since then (willing to bisect if necessary though).  

Messages thrown to terminal during compilation start with:
```
from /home/tester/Desktop/yuzu/yuzu/externals/dynarmic/src/backend/x64/a32_interface.cpp:9:
/usr/include/boost/icl/type_traits/type_to_string.hpp:56:12: error: partial specialization of ‘struct boost::icl::type_to_string<Unary<Type> >’ after instantiation of ‘struct boost::icl::type_to_string<std::__cxx11::basic_string<char> >’ [-fpermissive]
     struct type_to_string<Unary<Type> >
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated due to -Wfatal-errors.
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -179,9 +179,9 @@ set_property(DIRECTORY APPEND PROPERTY
 # System imported libraries
 # ======================
 
-find_package(Boost 1.64.0 QUIET)
+find_package(Boost 1.66.0 QUIET)
 if (NOT Boost_FOUND)
-    message(STATUS "Boost 1.64.0 or newer not found, falling back to externals")
+    message(STATUS "Boost 1.66.0 or newer not found, falling back to externals")
 
     set(BOOST_ROOT "${PROJECT_SOURCE_DIR}/externals/boost")
     set(Boost_NO_SYSTEM_PATHS OFF)

